### PR TITLE
[release-1.3] ⚠️ Use Kubernetes 1.26 in Quick Start docs and CAPD

### PR DESCRIPTION
**Pull request by aniruddha2000**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
part of #7671 

cc @kubernetes-sigs/cluster-api-release-team

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm

**Review by killianmuldoon — APPROVED** [review]
/lgtm

File: Tiltfile [hunk 1]
```diff
@@ -3,7 +3,8 @@
 envsubst_cmd = "./hack/tools/bin/envsubst"
 clusterctl_cmd = "./bin/clusterctl"
 kubectl_cmd = "kubectl"
-kubernetes_version = "v1.25.3"
+default_build_engine = "docker"
```
Comment (sbueringer) [inline Tiltfile:6]:
```suggestion
```
This feature is not implemented on release-1.3

File: test/e2e/clusterctl_upgrade_test.go [hunk 2]
```diff
@@ -66,9 +66,24 @@ var _ = Describe("When testing clusterctl upgrades (v1.2=>current)", func() {
...
-			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.2.5/clusterctl-{OS}-{ARCH}",
+			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.2.7/clusterctl-{OS}-{ARCH}",
 			InitWithProvidersContract: "v1beta1",
-			InitWithKubernetesVersion: "v1.25.3",
+			InitWithKubernetesVersion: "v1.26.0",
+		}
+	})
+})
+
+var _ = Describe("When testing clusterctl upgrades (v1.3=>current)", func() {
```
Comment (sbueringer) [inline test/e2e/clusterctl_upgrade_test.go:76]: Please drop this test case (we don't need v1.3 => release-1.3)

File: test/e2e/clusterctl_upgrade_test.go [hunk 3]
```diff
@@ -81,9 +96,25 @@ var _ = Describe("When testing clusterctl upgrades using ClusterClass (v1.2=>cur
...
+			InitWithBinary:            "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.2.7/clusterctl-{OS}-{ARCH}",
 			InitWithProvidersContract: "v1beta1",
-			InitWithKubernetesVersion: "v1.25.3",
+			InitWithKubernetesVersion: "v1.26.0",
+			WorkloadFlavor:            "topology",
+		}
+	})
+})
+
+var _ = Describe("When testing clusterctl upgrades using ClusterClass (v1.3=>current) [ClusterClass]", func() {
```
Comment (sbueringer) [inline test/e2e/clusterctl_upgrade_test.go:107]: Please drop this test case (we don't need v1.3 => release-1.3)

**Review by sbueringer — COMMENTED** [review]
@aniruddha2000 Thx, just a few nits

**Comment by fabriziopandini** [comment]
/hold 
after tomorrow's patch release; this will slit to next month's patch release (we need CI signal for it first)

**Comment by sbueringer** [comment]
Thx!

/lgtm

**Pull request merged**